api/middleware/verifyToken.js

```diff
@@ -0,0 +1,25 @@
+const pool = require('../configuration/config')
+const  jwt = require('jsonwebtoken')
+require('dotenv').config()
+
+const verifyToken = async (request, response, next) =>{
+const token = request.headers['x-access-token']
+if(!token){
+  return response.status(403).json({message:'Unathorized!'});
+}
+try {
+   const decoded = await jwt.verify(token , process.env.TOKEN_SECRET);
+  const text = 'SELECT * FROM users WHERE id = $1';   
+   const {rows} = await pool.query(text, [decoded.userId]);
+   if(!rows[0]){
+      return response.status(400).status({message: 'Token id invalid'})
+   }
```

Comment:
check here for how the `jwt.verify` works https://www.npmjs.com/package/jsonwebtoken

`jwt.verify` throws an error if the token is invalid, so you need to wrap it in a `try{}catch(error){}`.